packages/client/src/blockchain/chain.ts

```diff
@@ -2,14 +2,108 @@ import { Block, BlockHeader } from '@ethereumjs/block'
 import { Blockchain } from '@ethereumjs/blockchain'
 import { ConsensusAlgorithm, Hardfork } from '@ethereumjs/common'
 import { equalsBytes } from '@ethereumjs/util'
+import * as lmdb from 'lmdb'
 
 import { LevelDB } from '../execution/level'
 import { Event } from '../types'
 
 import type { Config } from '../config'
-import type { DB, DBObject, GenesisState } from '@ethereumjs/util'
+import type { BatchDBOp, DB, DBObject, EncodingOpts, GenesisState } from '@ethereumjs/util'
 import type { AbstractLevel } from 'abstract-level'
 
+export class LMDB<
+  TKey extends Uint8Array | string = Uint8Array | string,
+  TValue extends Uint8Array | string | DBObject = Uint8Array | string | DBObject
+> implements DB<TKey, TValue>
+{
+  private path
+  private database
+
+  // It's necessary to use a different path for every init to avoid reading old data from a previous test
+  static nonce: number
+
+  constructor(path?: string) {
+    if (LMDB.nonce === undefined) {
+      LMDB.nonce = 0
+    }
+    this.path = path ?? `./${LMDB.nonce++}`
+    this.database = lmdb.open({
+      compression: true,
+      name: '@ethereumjs/trie',
+      path: this.path,
+    })
+  }
+
+  /**
+   * @inheritDoc
+   */
+  async get(key: TKey): Promise<TValue | undefined> {
+    let value
+
+    // console.log('inside lmdb get')
+    // console.log(key)
+
+    try {
+      value = await this.database.get(key)
+      // console.log('value from get')
+      // console.log(value)
+      // console.log(value.toString('hex'))
+      if (value === null) return undefined
+    } catch (error: any) {
+      // https://github.com/Level/abstract-level/blob/915ad1317694d0ce8c580b5ab85d81e1e78a3137/abstract-level.js#L309
+      // This should be `true` if the error came from LevelDB
+      // so we can check for `NOT true` to identify any non-404 errors
+      if (error.notFound !== true) {
+        throw error
+      }
+    }
+    // eslint-disable-next-line
+    if (value instanceof Buffer) value = Uint8Array.from(value)
+
+    return value as TValue
+  }
+
+  /**
+   * @inheritDoc
+   */
+  async put(key: TKey, val: TValue): Promise<void> {
+    // console.log('inside lmdb put')
+    // console.log(key)
+
+    await this.database.put(key, val)
+  }
+
+  /**
+   * @inheritDoc
+   */
+  async del(key: TKey): Promise<void> {
+    await this.database.remove(key)
+  }
+
+  /**
+   * @inheritDoc
+   */
+  async batch(opStack: BatchDBOp<TKey, TValue>[]): Promise<void> {
+    const levelOps = []
+    for (const op of opStack) {
+      levelOps.push({ ...op })
+    }
+
+    // TODO: Investigate why as any is necessary
+    await this.database.batch(levelOps as any)
+  }
+
+  // not implemented
+  shallowCopy(): DB<TKey, TValue> {
+    // @ts-ignore
+    return
+  }
+
+  async open() {}
+
+  async close() {}
+}
+
```

Comment:
Not sure if this is meant to be permanently here, but this should be in a separate file `lmdb.ts`.

Reply:
(just to note, if we go this way).